### DISCUSSION TITLE
How to dispatch specific workflow

### DISCUSSION THREAD

**Discussion by hpposch** [elsa-workflows/elsa-core#1535] [Q&A]
Hi all!
I am currently working with custom activities, and i wrote myself a blocking activity.
Everything works as expectet.
I have created a controller which dispatches all workflows after it get's triggered.

Now i was wondering if it is possible to trigger a certain workflow. For example via the correlationId.
In my example i have 3 workflows in state 'Suspended' 
When i now call for example https://localhost:5001/files all workflows would be executed, because inside my controller there is the following code:
`[HttpPost]
        public async Task<IActionResult> Handle()
        {
            var collectedWorkflows = await _invoker.DispatchWorkflowsAsync();
            return Ok(collectedWorkflows.ToList());
        }`
So what can i do to trigger only one workflow that is in state 'Suspended'.
All other workflows should be untouched.

Thanks in advance for your help!
Heinz Peter

**Comment by sfmskywalker** [comment]
Hi @hpposch ,

You could use the higher-level `IWorkflowLaunchpad` service to find & resume specific workflow instances based on correlation ID or workflow instance ID if you have one.

For example, if you have a suspended workflow with correlation ID `myCorrelationId` then you can resume it as follows:

```
var query = new WorkflowQuery
{
   ActivityType = "MyCustomActivityType",
   CorrelationId: myCorrelationId
};

// Collect workflow instance IDs to resume.
var collectedWorkflows= await _workflowLaunchpad.FindWorkflowsAsync(query); // Will return all workflow instance IDs that are blocked on "MyCustomActivityType" and with the specified correlation ID.

// Resume collected workflow instance IDs.
var results = await _workflowLaunchpad.DispatchPendingWorkflowsAsync(collectedWorkflows);
```

**Reply by hpposch** [reply]
Thank you, that worked great 😊
